src/shared.c

```diff
@@ -602,14 +603,311 @@ void hc_string_trim_trailing (char *s)
   s[new_len] = 0;
 }
 
-size_t hc_fread (void *ptr, size_t size, size_t nmemb, FILE *stream)
+#if defined (__CYGWIN__)
+// workaround for zlib with cygwin build
+int _wopen(const char *path, int oflag, ...)
```

Comment:
Can you explain why you need this? Please also fix indents etc.

Reply:
it's the hack to make zlib build against cygwin !